crates/turbopack-dev/js/src/runtime.js

```diff
@@ -209,12 +210,42 @@ externalRequire.resolve = (name, opt) => {
   return require.resolve(name, opt);
 };
 
+/** @type {Map<ModuleId, Promise<any> | true>} */
+const availableModules = new Map();
+
 /**
  * @param {SourceInfo} source
- * @param {string} chunkPath
- * @returns {Promise<any> | undefined}
+ * @param {ChunkData} chunkData
+ * @returns {Promise<any>}
  */
-async function loadChunk(source, chunkPath) {
+async function loadChunk(source, chunkData) {
+  if (typeof chunkData === "string") {
+    return loadChunkPath(source, chunkData);
+  } else {
+    const includedList = chunkData.included || [];
+    const promises = includedList.map((included) => {
+      if (moduleFactories[included]) return true;
+      return availableModules.get(included);
+    });
+    if (promises.length > 0 && promises.every((p) => p)) {
+      // When all included items are already loaded or loading, we can skip loading ourselves
+      return Promise.all(promises);
+    }
+    const promise = loadChunkPath(source, chunkData.path);
+    for (const included of includedList) {
+      if (!availableModules.has(included))
+        availableModules.set(included, promise);
```

Comment:
The two promises should race.

1. Load chunk A with modules 1, 2
2. Load chunk B with modules 2, 3
3. Chunk B loads 
4. Load chunk C with module 2
5. Chunk A loads

In the current situation, chunk `C` will only be marked as loaded once `A` loads, but it should actually be marked as loaded instantly.